kwok/tests/usersignup.go

```diff
@@ -0,0 +1,65 @@
+package kwok
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+func PodmanBuildImage() (string, error) {
+	buildCmd := exec.Command("podman", "build", "--build-arg", "CLUSTER_NAME=test", "-t", "my-kwokctl-oc-image", "-f", "../Dockerfile", "../../.")
+	output, err := buildCmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to build image: %w", err)
+	}
+
+	fmt.Printf("BuildImage output: %s\n", output)
+	outputString := strings.Split(strings.TrimSpace(string(output)), "\n")
+	imageID := outputString[len(outputString)-1]
+
+	return imageID, nil
+}
+
+func PodmanRunContainer(imageID string) (string, error) {
+	cmd := exec.Command("podman", "run", "-d", "--rm", imageID)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to start command: %w", err)
+	}
+
+	containerID := strings.TrimSpace(string(output))
+
+	return containerID, nil
+}
+
+func PodmanExecCommand(containerID string, command string) (map[string]int64, error) {
```

Comment:
I think it might be worth documenting that the provided command must be one that is expected to return json output.